providers/cncf/kubernetes/src/airflow/providers/cncf/kubernetes/utils/xcom_sidecar.py

```diff
@@ -64,3 +73,27 @@ def add_xcom_sidecar(
     pod_cp.spec.containers.append(sidecar)
 
     return pod_cp
+
+
+def add_sidecar_to_spark_operator_pod_spec(
+    spec: dict, sidecar_container_image: str | None = None, sidecar_container_resources: dict | None = None
+):
```

Comment:
```suggestion
def add_sidecar_to_spark_operator_pod_spec(
    spec: dict[str, Any], sidecar_container_image: str | None = None, sidecar_container_resources: dict[str, Any] | None = None
) -> dict[str, Any]:
```

please confirm whether these type are correct. Thanks!